### PR TITLE
Add list to skip unnecessary rebase

### PR DESCRIPTION
**Pull request by jakobherdt**

### CODE REVIEW

**Comment by beatngu13** [comment]
Generally, a few commit messages are too long as you can see on the trailing ellipsis (…). Use at most 50 characters (as mentioned in [this blog post](https://chris.beams.io/posts/git-commit/)).

File: pom.xml [hunk 1]
```diff
@@ -50,18 +49,27 @@
 			<artifactId>commons-io</artifactId>
 			<version>2.1</version>
 		</dependency>
-
+		<dependency>
```
Comment (modulo11) [inline pom.xml]: This looks like an unused dependency.

File: src/main/java/org/retest/rebazer/service/BitbucketService.java [hunk 2]
```diff
@@ -33,31 +33,58 @@
...
+				merge(pullRequest);
+				pullRequestUpdateStates.remove(pullRequest.getId());
+			}
+		} else {
+			log.info("PR{} is unchanged since last run, last change at {}", pullRequest,
+					pullRequestUpdateStates.get(pullRequest.getId()));
+		}
+	}
+
+	boolean isChangedSinceLastRun(PullRequest pullRequest) {
```
Comment (beatngu13) [inline src/main/java/org/retest/rebazer/service/BitbucketService.java]: Should be `hasChanged…`.

File: src/main/java/org/retest/rebazer/service/BitbucketService.java [hunk 3]
```diff
@@ -33,31 +33,58 @@
...
-				} else {
-					merge(pullRequest);
-				}
-			});
+			getAllPullRequestIds(repo).forEach(pullRequest -> handlePR(repo, pullRequest));
 		});
 	}
 
+	private void handlePR(Repository repo, PullRequest pullRequest) {
+		log.info("Processing " + pullRequest);
```
Comment (beatngu13) [inline src/main/java/org/retest/rebazer/service/BitbucketService.java]: Shouldn't be on `INFO` level.

File: src/main/java/org/retest/rebazer/service/BitbucketService.java (hunk 2 above)
Comment (beatngu13) [inline src/main/java/org/retest/rebazer/service/BitbucketService.java]: Should be `hasChangedSinceLastRun(PullRequest)`.

File: src/test/java/org/retest/rebazer/service/BitbucketServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		bitbucketLegacyTemplate = mock(RestTemplate.class);
+		config = mock(RebazerProperties.class);
+		rebaseService = mock(RebaseService.class);
+		pullRequestUpdateStates = new HashMap<>();
+		cut = new BitbucketService(bitbucketTemplate, bitbucketLegacyTemplate, config, rebaseService,
+				pullRequestUpdateStates);
+	}
+
+	@Test
+	public void isChangedSinceLastRun_should_return_false_if_pullrequest_didnt_change() {
```
Comment (beatngu13) [inline src/test/java/org/retest/rebazer/service/BitbucketServiceTest.java]: Adapt test names according to the method name change.

File: src/main/java/org/retest/rebazer/domain/PullRequest.java [hunk 5]
```diff
@@ -1,5 +1,7 @@
 package org.retest.rebazer.domain;
 
+import java.util.Date;
```
Comment (beatngu13) [inline src/main/java/org/retest/rebazer/domain/PullRequest.java]: Obsolete.

File: src/main/java/org/retest/rebazer/service/BitbucketService.java (hunk 3 above)
Comment (beatngu13) [reply src/main/java/org/retest/rebazer/service/BitbucketService.java]: Still open.

File: src/test/java/org/retest/rebazer/service/BitbucketServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import org.retest.rebazer.domain.PullRequest;
+import org.springframework.web.client.RestTemplate;
+
+public class BitbucketServiceTest {
+
+	RestTemplate bitbucketTemplate;
+	RestTemplate bitbucketLegacyTemplate;
+	RebazerProperties config;
+	RebaseService rebaseService;
+	Map<Integer, String> pullRequestUpdateStates;
```
Comment (beatngu13) [inline src/test/java/org/retest/rebazer/service/BitbucketServiceTest.java]: Let's see which fields we really need for now.

**Pull request merged**